### DISCUSSION TITLE
How Dual-cluster upgrade mode transform the orgin pg data to target version structure?

### DISCUSSION THREAD

**Discussion by mali-nuist** [Kong/kong#13406] [Help]
I am currently reading the Kong documentation to understand the upgrade process.
In the "in-place mode," the "kong migrations up/finish" command is utilized for data migrations. 
However, I am struggling to understand how to handle PostgreSQL data migration in the "dual-cluster upgrade mode." 
The documentation only mentions restoring original data to the new database but seems lack the process of migrating data structures to the target version schema.
Could anyone can provide some further explanation? Thank you!
![image](https://github.com/user-attachments/assets/27ae45d1-96b4-4039-a0a8-674f0ca74099)

**Comment by xianghai2** [comment]
@outsinre could you help look into this? thanks a lot.
